server/routes/linkTokens.js

```diff
@@ -28,7 +28,40 @@ router.post(
   '/',
   asyncWrapper(async (req, res) => {
     try {
-      const { userId, transferIntentId } = req.body;
+      const { userId, subscriptionAmount } = req.body;
+      const { username: username } = await retrieveUserById(userId);
+      const transIntentCreateRequest = {
```

Comment:
Would it make sense to extract this code into two functions? So you might call something like,
```
const transferIntentId = createTransferIntent(userId, amount);
const linkTokenResponse = createLinkTokenForTransfer(transferIntentId, userId, amount);
```
It might make it clearer what the two steps are in this process

Reply:
good idea!